### PR TITLE
Gh 2249 accumulo upgrade to 1.9.3

### PR DESCRIPTION
**Pull request by macenturalxl1**
Hi - I've made the changes for Accumulo 1.9.3:

- I've added new tests for the `rfilereaderrdd` package in JUnit5 to help increase the coverage and to manage any changes in behaviour after upgrading
- Also refactored some of the existing tests and upgraded to JUnit5 to use some of the newer features
- I fixed the compiler errors RFileReaderIterator caused by updating the Accumulo version to 1.9.3
- I ran the integration tests locally against a MiniAccumulo cluster I ran locally and all passed
- I also updated the README how to set up integration tests on a local cluster to help future developers/users

I can add more tests and increase the coverage in the modules that consume the Accumulo dependency? - I wasn't sure how far to go with it so thought I'd raise the pull request anyway and see what you think - I'm don't mind writing more.

Also let me know of any other recommendations/changes you require for this or if you run any additional Accumulo tests of your own that picks something? 

Cheers

### CODE REVIEW

**Comment by macenturalxl1** [comment]
I've seen the Checkstyle failed on spark-accumulo-library from the Travis build so will fix this now...

**Comment by t11947** [comment]
Could you sign the Contributor License Agreement?

**Comment by macenturalxl1** [comment]
Ah I've ran the tests in the IDE that passed and not ran the ./cd/verify.sh script instead where I can see the failed tests. Will fix this now too.
I'll sign the CLA too

File: library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIteratorTest.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+
+public class RFileReaderIteratorTest {
+
+    @Test
+    @DisplayName("Init RFileReaderIterator with invalid AccumuloTablet file throws RunTimeException")
```
[the remark this answers is not in the document]
Comment (macenturalxl1) [reply library/spark/spark-accumulo-library/src/test/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/rfilereaderrdd/RFileReaderIteratorTest.java]: Yes that's a typo thanks, corrected it

**Comment by macenturalxl1** [comment]
There's a test that are failing on the pipeline from flink-library package: 
`AddElementsFromKafkaHandlerIT shouldAddElementsWithByteArrayConsumer()`

When I try to replicate it locally on the develop branch, I'm getting the same failing test. Is this expected? Just exploring ways to fix the test

**Comment by macenturalxl1** [comment]
Hi I've not made any change to the flink-library test and it passed 3rd time - do you think this should be recorded as an new Issue?

**Comment by d47853** [comment]
As long as it wasn't caused as a result of this change. Do you have any logs or stack traces which could be used in debugging? removing the -q option should print the output.

**Comment by macenturalxl1** [comment]
> As long as it wasn't caused as a result of this change. Do you have any logs or stack traces which could be used in debugging? removing the -q option should print the output.

I haven't made a change to the module or its pom.xml - I can't get it to fail now after running it a few times

**Comment by t11947** [comment]
> > As long as it wasn't caused as a result of this change. Do you have any logs or stack traces which could be used in debugging? removing the -q option should print the output.
> 
> I haven't made a change to the module or its pom.xml - I can't get it to fail now after running it a few times

I'd say open it so that we have a log of this issue should it be encountered in the future. We'll just make sure it's labelled appropriately.

**Comment by macenturalxl1** [comment]
Hi got it to fail - here are the logs attached
[flink-library-logs.txt](https://github.com/gchq/Gaffer/files/4369699/flink-library-logs.txt)

**Comment by t11947** [comment]
After speaking with the team, we've decided that we'd like to upgrade to JUnit 5 in a different PR and upgrade everything at once. Would you be mind using JUnit 4 for this PR?

**Comment by macenturalxl1** [comment]
> After speaking with the team, we've decided that we'd like to upgrade to JUnit 5 in a different PR and upgrade everything at once. Would you be mind using JUnit 4 for this PR?

Sure, I can downgrade the exisiting ones to JUnit4

**Comment by macenturalxl1** [comment]
Hi - I've made the changes to Junit4. I thought I'd raise the change I made in `GetRDDOfAllElementsHandlerIT` - I've kept parameterisation to help make the _individual_ tests more readable, but because I've implemented it the JUnit4 way and not sure if _overall_ the use of it outweighs readibility of the class - so was in two minds of reverting it to original or keeping parameterisation to help migrate it to JUnit5 at a later date like you mentioned.
I'll let you make the call whichever you prefer

**Comment by macenturalxl1** [comment]
Hi - is this error something you have ever come across before? I've tried looking at a few solutions and haven't got anywhere with it:
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project accumulo-store: ExecutionException Error occurred in starting fork`

It something that occurs on the branch I'm working on only. All my tests pass. Even when I @Ignore and //comment out the tests i've made changes to it still occurs as I have considered rolling back the test files I've made changes in.

**Comment by t11947** [comment]
> Hi - is this error something you have ever come across before? I've tried looking at a few solutions and haven't got anywhere with it:
> `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project accumulo-store: ExecutionException Error occurred in starting fork`
> There's no more logs to say how, why or where the ExecutionExecution was caused - even when using -e and -X for more logging.
> 
> It something that occurs on the branch I'm working on only. All my tests pass. Even when I @ignore and //comment out the tests i've made changes to it still occurs as I have considered rolling back the test files I've made changes in - so can't see where I have caused it. It's the first time I've come across it after downgrading to JUnit4

This isn't something I've encountered before and it sounds like a tricky one. I had a quick look around for solutions online, but as is often the case, there are myriad potential fixes, but rarely an explanation as to what was causing the issue. Do the tests pass locally and only fail on Travis?

**Comment by macenturalxl1** [comment]
> > Hi - is this error something you have ever come across before? I've tried looking at a few solutions and haven't got anywhere with it:
> > `[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project accumulo-store: ExecutionException Error occurred in starting fork`
> > There's no more logs to say how, why or where the ExecutionExecution was caused - even when using -e and -X for more logging.
> > It something that occurs on the branch I'm working on only. All my tests pass. Even when I @ignore and //comment out the tests i've made changes to it still occurs as I have considered rolling back the test files I've made changes in - so can't see where I have caused it. It's the first time I've come across it after downgrading to JUnit4
> 
> This isn't something I've encountered before and it sounds like a tricky one. I had a quick look around for solutions online, but as is often the case, there are myriad potential fixes, but rarely an explanation as to what was causing the issue. Do the tests pass locally and only fail on Travis?

That's what I found with possible solutions online - couldn't find what could be the cause.
The tests appear to pass both locally and on Travis as the results say there were no failures or errors - both give identical logs - I'll see if I can get any further with it.

**Comment by macenturalxl1** [comment]
Hi I found the cause of what was failing sure-fire, it was my change in UpdateTableIterator test suite so I just reverted the class

File: store-implementation/accumulo-store/src/test/resources/empty-store.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright 2016-2020 Crown Copyright
```
Comment (d47853) [inline store-implementation/accumulo-store/src/test/resources/empty-store.properties]: Any new files you create (this one, RFileReaderIteratorTest.java and AccumuloTabletTest.java should have just the current year in the copyright. For example
```suggestion
# Copyright 2020 Crown Copyright
```

Comment (macenturalxl1) [reply store-implementation/accumulo-store/src/test/resources/empty-store.properties]: Thanks and done

**Comment by d47853** [comment]
If you're able to, could you merge develop into your branch?

**Comment by macenturalxl1** [comment]
> If you're able to, could you merge develop into your branch?

Done, triggered a new build

**Pull request merged**